inventory/group_vars/all.yml

```diff
@@ -215,7 +215,9 @@ nginx_sites:
 
       error_page 500 502 503 504 /500.html;
       client_max_body_size 4G;
-      keepalive_timeout 60;
+      keepalive_timeout {{ unicorn_timeout }};
+      proxy_read_timeout {{ unicorn_timeout }};
+      proxy_send_timeout {{ unicorn_timeout }};
```

Comment:
doesn't this mean that nginx will have `unicorn_timeout` to send data to unicorn and the same `unicorn_timeout` to read the response. Am I right? is that what we need?

Reply:
I'm not sure.

I think, the send is usually quick and then we need a lot of time for the read. But we don't actually know. The product import could be very different.

For example, if the timeout is one minute then nginx could send data for one minute and then read data for one minute before it times out. But I guess that unicorn would time out after the first minute and we see an error message. I don't know though if unicorn starts counting after the read or before.

Anyway, this configuration ensures that nginx is always waiting at least as long as unicorn. We don't want nginx to timeout earlier and process the next request while the unicorn worker is still busy. In that case we would waste time on a worker and requests would pile up. So it's okay or even good if nginx has a higher timeout than unicorn (send plus read).

Reply:
it's something worth finding out eventually.